### PR TITLE
Add enrich processor and APIs

### PR DESCRIPTION
**Pull request by russcam**
Relates: #4232

This PR adds the ingest enrich processor and the enrich APIs.

Patch Get enrich policy API to set name URL part as list type.

Add Enrich to XPackUsage and XPackInfo. Obsolete DataFrame and introduce Transform,
and conditionally check to allow integration tests to pass.

### CODE REVIEW

File: src/Nest/XPack/Enrich/Stats/EnrichStatsRequest.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+using System;
+using System.Runtime.Serialization;
+
+namespace Nest
+{
+	/// <summary>
+	/// Deletes an enrich policy
```
Comment (codebrain) [inline src/Nest/XPack/Enrich/Stats/EnrichStatsRequest.cs]: Copy-pasta? Naming seems a little odd...

File: src/Nest/XPack/Info/XPackInfo/XPackInfoResponse.cs [hunk 2]
```diff
@@ -78,17 +88,27 @@ public class XPackFeatures
 		[DataMember(Name = "security")]
 		public XPackFeature Security { get; internal set; }
 
+		// TODO! Expand to fullname in 8.0?
```
Comment (codebrain) [inline src/Nest/XPack/Info/XPackInfo/XPackInfoResponse.cs]: Can probably delete this TODO

File: src/Nest/XPack/Enrich/Stats/EnrichStatsRequest.cs (hunk 1 above)
Comment (russcam) [reply src/Nest/XPack/Enrich/Stats/EnrichStatsRequest.cs]: yeah, weird! Will fix

File: src/Nest/XPack/Info/XPackInfo/XPackInfoResponse.cs (hunk 2 above)
Comment (russcam) [reply src/Nest/XPack/Info/XPackInfo/XPackInfoResponse.cs]: Thanks, will remove

**Comment by russcam** [comment]
ported to master

**Pull request merged**